src/mongo/db/auth/external/sasl_oidc_server_mechanism.cpp

```diff
@@ -120,6 +142,11 @@ StatusWith<std::tuple<bool, std::string>> SaslOidcServerMechanism::step1(
             fmt::format("No identity provider found for principal name `{}`", *principalName)};
```

Comment:
Will dereference work correctly if principalName is boost::none?

Reply:
Fixed